commitrmnocb/report.go

```diff
@@ -0,0 +1,84 @@
+package commitrmnocb
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/smartcontractkit/libocr/offchainreporting2plus/ocr3types"
+
+	cciptypes "github.com/smartcontractkit/chainlink-common/pkg/types/ccipocr3"
+
+	"github.com/smartcontractkit/chainlink-ccip/commit"
+)
+
+func (p *Plugin) Reports(seqNr uint64, outcomeBytes ocr3types.Outcome) ([]ocr3types.ReportWithInfo[[]byte], error) {
+	outcome, err := DecodeCommitPluginOutcome(outcomeBytes)
+	if err != nil {
+		// TODO: metrics
+		p.lggr.Errorw("failed to decode Outcome", "outcomeBytes", outcomeBytes, "err", err)
+		return nil, fmt.Errorf("failed to decode Outcome: %w", err)
+	}
+
+	if outcome.OutcomeType != ReportGenerated {
+		return []ocr3types.ReportWithInfo[[]byte]{}, nil
+	}
+
+	rep := cciptypes.NewCommitPluginReport(outcome.RootsToReport, outcome.TokenPrices, outcome.GasPrices)
+
+	encodedReport, err := p.reportCodec.Encode(context.Background(), rep)
+	if err != nil {
+		return nil, fmt.Errorf("encode commit plugin report: %w", err)
+	}
+
```

Comment:
A log here might be useful as well, of the pre-encoded report `rep`

Reply:
We log the report in `ShouldTransmitAcceptedReport`, I think that's sufficient.